STX.REST.RESTFulSense.Clients/Services/Foundations/HttpExchanges/HttpExchangeService.Validations.cs

```diff
@@ -0,0 +1,204 @@
+// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Net;
+using System.Net.Http;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges;
+using STX.REST.RESTFulSense.Clients.Models.Services.HttpExchanges.Exceptions;
+
+namespace STX.REST.RESTFulSense.Clients.Services.Foundations.HttpExchanges
+{
+    internal partial class HttpExchangeService
+    {
+        private static void ValidateOnGet(
+            HttpExchange httpExchange,
+            HttpMethod defaultHttpMethod,
+            Version defaultHttpVersion,
+            HttpVersionPolicy defaultHttpVersionPolicy)
+        {
+            Validate(
+                (Rule: IsInvalid(httpExchange), Parameter: nameof(HttpExchange)),
+                (Rule: IsInvalid(defaultHttpMethod), Parameter: nameof(HttpMethod)),
+                (Rule: IsInvalid(defaultHttpVersion), Parameter: nameof(Version)),
+                (Rule: IsInvalid(defaultHttpVersionPolicy), Parameter: nameof(HttpVersionPolicy)));
+
+            ValidateHttpExchange(
+                httpExchange,
+                defaultHttpMethod,
+                defaultHttpVersion,
+                defaultHttpVersionPolicy);
+        }
+
+        private static void ValidateHttpExchange(
+            HttpExchange httpExchange,
+            HttpMethod httpMethod,
+            Version httpVersion,
+            HttpVersionPolicy httpVersionPolicy)
+        {
+            ValidateHttpExchangeRequest(
+                httpExchangeRequest: httpExchange.Request,
+                httpMethod,
+                httpVersion,
+                httpVersionPolicy);
+        }
+
+        private static void ValidateHttpExchangeRequest(
+            HttpExchangeRequest httpExchangeRequest,
+            HttpMethod httpMethod,
+            Version httpVersion,
+            HttpVersionPolicy httpVersionPolicy)
+        {
+            ValidateHttpExchangeRequestNotNull(httpExchangeRequest);
+
+            ValidateInputArgument(
+                (Rule: IsInvalid(httpExchangeRequest.BaseAddress),
+                    Parameter: nameof(HttpExchangeRequest.BaseAddress)),
+
+                (Rule: IsInvalid(httpExchangeRequest.RelativeUrl),
+                    Parameter: nameof(HttpExchangeRequest.RelativeUrl)),
+
+                (Rule: IsInvalidHttpMethod(httpExchangeRequest.HttpMethod, httpMethod),
+                    Parameter: nameof(HttpExchangeRequest.HttpMethod)),
+
+                (Rule: IsInvalidHttpVersion(httpExchangeRequest.Version, httpVersion),
+                    Parameter: nameof(HttpExchangeRequest.Version)),
+
+                (Rule: IsInvalidHttpVersionPolicy(httpExchangeRequest.VersionPolicy, httpVersionPolicy),
+                    Parameter: nameof(HttpExchangeRequest.VersionPolicy)));
```

Comment:
Don't use ValidateInputArgument.  This is not correct.  You are not validating method parameters / arguments.

Use a new validate method  `ValidateHttpExchangeRequestObject` and this new validate method should throw `InvalidHttpExchangeRequestException`

```suggestion
            ValidateHttpExchangeRequestObject(
                (Rule: IsInvalid(httpExchangeRequest.BaseAddress),
                    Parameter: nameof(HttpExchangeRequest.BaseAddress)),

                (Rule: IsInvalid(httpExchangeRequest.RelativeUrl),
                    Parameter: nameof(HttpExchangeRequest.RelativeUrl)),

                (Rule: IsInvalidHttpMethod(httpExchangeRequest.HttpMethod, httpMethod),
                    Parameter: nameof(HttpExchangeRequest.HttpMethod)),

                (Rule: IsInvalidHttpVersion(httpExchangeRequest.Version, httpVersion),
                    Parameter: nameof(HttpExchangeRequest.Version)),

                (Rule: IsInvalidHttpVersionPolicy(httpExchangeRequest.VersionPolicy, httpVersionPolicy),
                    Parameter: nameof(HttpExchangeRequest.VersionPolicy)));
```